pi5/pinctrl_utils.go

```diff
@@ -63,6 +74,34 @@ func (b *pinctrlpi5) findPathFromAlias(nodeName string) (string, error) {
 	return nodePath, err
 }
 
+// read in 'numCells' 32 bit chunks from byteContents, the bytestream outputted from reading the file '/ranges'. Convert bytes into their uint64 value
+func parseCells(numCells uint32, byteContents *[]byte) (uint64, error) {
+	var parsedValue uint64
+
+	if len(*byteContents) < int(numCells)*4 {
+		errorMsg := "num cells was: " + string(numCells) + ", but there aren't enough bytes to read in from inputted bytestream"
+		return 0, errors.New(errorMsg)
+	}
+
+	switch numCells {
+
+	// reading in 32 bits. regardless we must convert to a 64 bit address so we add a bunch of 0s to the beginning.
+	case 1:
+		parsedValue = uint64(binary.BigEndian.Uint32((*byteContents)[:(4 * numCells)]))
+
+	// reading in 64 bits
+	case 2:
+		parsedValue = binary.BigEndian.Uint64((*byteContents)[:(4 * numCells)])
```

Comment:
in case 1 -> you're given a 32 bit integer, and you must change it to a uint64 since addresses are represented by 64 bits. after converting our 32bit number using bigEndian conversion, we effectively  add a bunch of 0s to the beginning using the uint64 cast.

in case 2 -> we have a 64 bit number being read in to its very easy to use bigEndian conversion to just convert it straight into a uint64